packages/gestalt/src/InternalDismissButton.js

```diff
@@ -0,0 +1,108 @@
+// @flow strict
+import { type Node, forwardRef, useImperativeHandle, useState, useRef } from 'react';
+import classnames from 'classnames';
+import Pog from './Pog.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+import styles from './IconButton.css';
+import touchableStyles from './Touchable.css';
+import useFocusVisible from './useFocusVisible.js';
+import useTapFeedback from './useTapFeedback.js';
+
+type Props = {|
+  accessibilityLabel: string,
+  accessibilityControls?: string,
+  onClick?: AbstractEventHandler<
+    SyntheticMouseEvent<HTMLButtonElement> | SyntheticKeyboardEvent<HTMLButtonElement>,
+  >,
+|};
+
+const InternalDismissIconButtonWithForwardRef: React$AbstractComponent<Props, HTMLButtonElement> =
```

Comment:
Can you add comments to this file why we're creating a custom button and not using IconButton?

Reply:
DONE